src/xref2/env.ml

```diff
@@ -372,18 +369,15 @@ let lookup_root_module name env =
         | Found u -> (
             match u.root.id with
             | `Root _ as id ->
-                let m =
-                  Component.Delayed.put (fun () ->
-                      let unit = r.resolve_unit u.root in
-                      module_of_unit unit)
-                in
-                Some (Resolved (u.root.digest, id, u.hidden, m))
+                let unit = r.resolve_unit u.root in
+                let m = module_of_unit unit in
+                Some (Resolved (u.root.digest, id, m))
```

Comment:
Resolving root modules is no longer lazy because we need to check for `canonical` everytime.
Is here a case where this would cause a slowdown ? (no change on base)

Reply:
Could be actually - could you do a test run on core_kernel or similar? I'd be interested to see the difference in memory usage too.

Reply:
I find no regression on core_kernel (29s in both cases). Maximum memory doesn't seem to change too.